### PR TITLE
[IMPROVE] Request user presence on demand

### PR DESCRIPTION
**Pull request by ggazzo**

### CODE REVIEW

File: app/api/server/v1/users.js [hunk 1]
```diff
@@ -659,6 +659,12 @@ API.v1.addRoute('users.presence', { authRequired: true }, {
 			},
 		};
 
+		if (ids) {
+			return API.v1.success({
+				users: Users.findNotOfflineByIds(Array.isArray(ids) ? ids : ids.split(','), options).fetch(),
```
Comment (sampaiodiego) [inline app/api/server/v1/users.js]:
```suggestion
				users: Users.findNotOfflineByIds(Array.isArray(ids) ? ids : ids.split(','), options).fetch(),
				full: false
```

File: app/ui-sidenav/client/sidebarItem.js [hunk 2]
```diff
@@ -201,6 +201,9 @@ Template.sidebarItem.events({
 });
 
 Template.sidebarItemIcon.helpers({
+	uid() {
+		return this.rid.replace(this.u._id, '');
```
Comment (sampaiodiego) [inline app/ui-sidenav/client/sidebarItem.js]: can this replace be done only if direct message?

File: imports/startup/client/listenActiveUsers.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import { Meteor } from 'meteor/meteor';
-import { Tracker } from 'meteor/tracker';
+// import { Tracker } from 'meteor/tracker';
```
Comment (sampaiodiego) [inline imports/startup/client/listenActiveUsers.js]:
```suggestion
```

File: imports/startup/client/listenActiveUsers.js [hunk 4]
```diff
@@ -72,18 +72,18 @@ const getUsersPresence = debounce(async (isConnected) => {
 	}
 }, 1000);
 
-let wasConnected = false;
-Tracker.autorun(() => {
-	if (!Meteor.userId() || !Meteor.status().connected) {
-		return;
-	}
+// let wasConnected = false;
```
Comment (sampaiodiego) [inline imports/startup/client/listenActiveUsers.js]: please remove this commented code

File: app/ui-sidenav/client/sidebarItem.js (hunk 2 above)
Comment (ggazzo) [reply app/ui-sidenav/client/sidebarItem.js]: hmmm I dont think we gonna have the "presence status" if its not a direct message, that why I didnt insert a test here

**Comment by ankar84** [comment]
I learned about such an **important innovation** only from the [YouTube video](https://youtu.be/fakyj0BQviQ?t=802) (so thankful for that!)
I am sure that **such extremely important changes require descriptions that are as detailed as possible**. 
Thanks a lot, anyway!

**Comment by EionRobb** [comment]
Is there any way a client can subscribe to status changes after this change, or is it now completely polling-only?

**Comment by emikolajczak** [comment]
Hi
I have a question about parameter `DISABLE_PRESENCE_MONITOR` and this change in presence monitor.
Will statuses show correctly when I set `DISABLE_PRESENCE_MONITOR` on `true`.
I am asking beacues set `DISABLE_PRESENCE_MONITOR` on `true` cause lower CPU load and better performance of Rocket.Chat.

**Comment by sampaiodiego** [comment]
they're not entirely related.. `DISABLE_PRESENCE_MONITOR` will prevent rocket.chat instances from monitoring (an updating) users's presence status.. while this PR just change how the client requests users' status from server.

in other words, if you use `DISABLE_PRESENCE_MONITOR` on all rocket.chat instances, all your users will appear as offline.

**Comment by ggazzo** [comment]
> Is there any way a client can subscribe to status changes after this change, or is it now completely polling-only?

the realtime data still arrives from `stream-notify-logged` "eventName":"user-status" we just changed how much info we loaded by the first time

**Comment by codeneno** [comment]
@ggazzo does this work on 2.4.12?

**Pull request merged**